### DISCUSSION TITLE
Errore npm

### DISCUSSION THREAD

**Discussion by mkel84** [indomus/forum#1377] [ZigBee, Z-Wave, Thread, Matter]
Salve,
oggi per errore ho aggiornato npm alla versione 8 e quando provo ad aggiornare zigbee attraverso il comando npm ci mi da il seguente errore:
npm does not support Node.js v10.15.1
You should probably upgrade to a newer version of node as we
can't make any promises that npm will work with this version.
You can find the latest version at https://nodejs.org/
/opt/zigbee2mqtt/lib/node_modules/npm/lib/npm.js:32
  #unloaded = false
  ^

SyntaxError: Invalid or unexpected token
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:657:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)

Come posso fare a tornare alla versione precedente?

**Comment by Scialla** [comment]
npm install -g npm@numeroversionechevuoiinstallare

**Reply by mkel84** [reply]
Purtroppo qualsiasi comando provo a dare con npm mi da l'errore postato in precedenza, mi tocca reinstallare tutto?

**Reply by Scialla** [reply]
dipende da quanto incasinato è il tuo sistema... puoi provare ad aggiornarlo, installare una versione aggiornata di nodejs, installare npm, sistemare eventuali conflitti che emergeranno; se ne sei in grado puoi provare a sistemare tutto un passo alla volta..

Se non ne sei in grado ti consiglio di salvarti la cartella di homeassistant, le configurazioni di eventuali broker mqtt, zigbee2mqtt, caddy o similari, insomma un bel backup e poi ti consiglio di migrare a homeassistant core su docker dopo aver piallato SD o SSD e reinstallato Buster, o meglio, Bullseye; così facendo non dovrai più essere tu a risolvere eventuali conflitti del sistema perché scaricherai una immagine di HA già aggiornata e funzionante; ci sono le guide su inDomus per fare tutto, anche per la migrazione.

Io ho appena migrato da Core su Venv a Core su Docker proprio perché avevo grandi conflitti di librerie che non riuscivo a risolvere
